rudder-lang/src/ast/enums.rs

```diff
@@ -6,115 +6,164 @@ use super::resource::Statement;
 use crate::error::*;
 use crate::parser::{PEnum, PEnumExpression, PSubEnum, Token};
 use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+// TODO enum alias
+// TODO 
+// - trouver les todo
+
+/// This item type is internal, because First and Fast cannot be constructed from an enum declaration or from and enum expression
+#[derive(Debug,Hash,PartialEq,Eq,Clone)]
+enum EnumItem<'src> {
+    First(Token<'src>), // token is the parent item
+    Item(Token<'src>),
+    Last(Token<'src>), // token is the parent item
+}
+
+/// Format the full token for compiler debug info
+impl<'src> fmt::Display for EnumItem<'src> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            EnumItem::First(p) => write!(f, "'First item of {}'", p),
+            EnumItem::Item(i) => write!(f, "{}", i),
+            EnumItem::Last(p) => write!(f, "'Last item of {}'", p),
+        }
+    }
+}
+
+/// Convert a Vec of tokns into a vec of EnumItems
+fn from_item_vec<'src>(father: Token<'src>, items: Vec<Token<'src>>) -> Vec<EnumItem<'src>> {
+        // detect incompleteness
+        let incomplete = items.contains(&Token::from("*"));
+        if incomplete {
+            let mut item_list = items.into_iter().filter(|x| *x != Token::from("*")).map(EnumItem::Item).collect::<Vec<EnumItem>>();
+            item_list.insert(0, EnumItem::First(father));
+            item_list.push(EnumItem::Last(father));
+            item_list
+        } else {
+            items.into_iter().map(EnumItem::Item).collect()   
+        }
+}
 
 // TODO Should a non global enum item be permitted as a variable name and must it be unique ?
 /// An enum tree is a structure containing the whole definition of an enum
 #[derive(Debug)]
-pub struct EnumTree<'src> {
+struct EnumTree<'src> {
     // a global tree has an automatic variable of its name and its items cannot be used as a variable name
```

Comment:
Comment is hard to understand (and to explain), maybe something like:
The global tree name is considered a declared (and global ?) variable name. The same goes for all of its items.

Reply:
except that your wording is false
items are not variable names